File: src/main/java/net/azzy/pulseflux/registry/damagetypes/DamageIncandescence.java
package net.azzy.pulseflux.registry.damagetypes;

import net.minecraft.entity.damage.DamageSource;

import static net.azzy.pulseflux.PulseFlux.MOD_ID;

public class DamageIncandescence extends DamageSource {

    public static final DamageIncandescence INCANDESCENCE = (DamageIncandescence) new DamageIncandescence().setBypassesArmor().setExplosive();

    protected DamageIncandescence() {
        super(MOD_ID + ".incandescence");
    }
}

File: src/main/java/net/azzy/pulseflux/registry/damagetypes/DamageIncandescence.kt
package net.azzy.pulseflux.registry.damagetypes

import net.minecraft.entity.damage.DamageSource

object DamageIncandescence : DamageSource {
    val INCANDESCENCE = DamageIncandescence().setBypassesArmor().setExplosive() as DamageIncandescence
}